File: backend/FinancialControl.Api/Models/RecurringTransaction.cs
namespace FinancialControl.Api.Models;

public class RecurringTransaction
{
    public Guid Id { get; set; }
    public Guid AccountId { get; set; }
    public Guid CategoryId { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public TransactionType Type { get; set; }
    public RecurrenceFrequency Frequency { get; set; }
    public int DayOfMonth { get; set; } // 1-31 para mensal, 1-7 para semanal
    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime? LastExecutionDate { get; set; }
    public DateTime? NextExecutionDate { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public Account Account { get; set; } = null!;
    public Category Category { get; set; } = null!;
}

public enum RecurrenceFrequency
{
    Daily = 1,
    Weekly = 2,
    Biweekly = 3,
    Monthly = 4,
    Quarterly = 5,
    Yearly = 6
}

File: frontend/src/types/recurringTransaction.ts
export type RecurrenceFrequency = 'Daily' | 'Weekly' | 'Biweekly' | 'Monthly' | 'Quarterly' | 'Yearly';

export type TransactionType = 'Income' | 'Expense';

export interface RecurringTransaction {
  id: string;
  accountId: string;
  categoryId: string;
  description: string;
  amount: number;
  type: TransactionType;
  frequency: RecurrenceFrequency;
  dayOfMonth: number;
  startDate: string;
  endDate?: string;
  isActive: boolean;
  lastExecutionDate?: string;
  nextExecutionDate?: string;
  createdAt: string;
  updatedAt: string;
  category?: {
    id: string;
    accountId: string;
    name: string;
    color: string;
    icon: string;
    type: string;
    createdAt: string;
  };
}

export interface CreateRecurringTransactionRequest {
  accountId: string;
  categoryId: string;
  description: string;
  amount: number;
  type: TransactionType;
  frequency: RecurrenceFrequency;
  dayOfMonth: number;
  startDate: string;
  endDate?: string;
}

export interface UpdateRecurringTransactionRequest {
  categoryId: string;
  description: string;
  amount: number;
  frequency: RecurrenceFrequency;
  dayOfMonth: number;
  endDate?: string;
  isActive: boolean;
}

export const FREQUENCY_LABELS: Record<RecurrenceFrequency, string> = {
  Daily: 'Diária',
  Weekly: 'Semanal',
  Biweekly: 'Quinzenal',
  Monthly: 'Mensal',
  Quarterly: 'Trimestral',
  Yearly: 'Anual',
};

export const FREQUENCY_ICONS: Record<RecurrenceFrequency, string> = {
  Daily: '📅',
  Weekly: '📆',
  Biweekly: '🗓️',
  Monthly: '📋',
  Quarterly: '📊',
  Yearly: '🎯',
};
